src/ImageSharp.Web/Providers/PhysicalFileSystemProvider.cs

```diff
@@ -71,5 +84,23 @@ public Task<IImageResolver> GetAsync(HttpContext context)
             var metadata = new ImageMetadata(fileInfo.LastModified.UtcDateTime, fileInfo.Length);
             return Task.FromResult<IImageResolver>(new PhysicalFileSystemResolver(fileInfo, metadata));
```

Comment:
I would recommend renaming `PhysicalFileSystemResolver` into `FileInfoImageResolver`, as it's resolving the image from an `Microsoft.Extensions.FileProviders.IFileInfo` and that's not specific to the `PhysicalFileProvider`.

Reply:
The type is explicitly using the `PhysicalFileProvider`

https://github.com/SixLabors/ImageSharp.Web/pull/214/files#diff-d17afca30cd91fe17418d73800c1e3003d61ac7364355e1fd1809acbdcc8def3R57-R59

Reply:
I'm not going to introduce a `WebRootProvider`. The current `PhysicalFileSystemProvider` is backwards compatible with out of the box behavior. Umbraco requires significant startup configuration to work as required already and writing providers is trivial enough..

Reply:
It's not only Umbraco that's using the `WebRootFileProvider` to provide access to static assets though! This change also breaks all [static assets referenced from a Razor Class Library](https://docs.microsoft.com/en-us/aspnet/core/razor-pages/ui-class?view=aspnetcore-6.0&tabs=visual-studio#consume-content-from-a-referenced-rcl-1) (during development/when not running the published application).

The default ASP.NET Core `WebHost` [calls `UseStaticWebAssets()` on a development environment](https://github.com/dotnet/aspnetcore/blob/c85baf8db0c72ae8e68643029d514b2e737c9fae/src/DefaultBuilder/src/WebHost.cs#L212-L220), which uses the same [`CompositeFileProvider` technique](https://github.com/dotnet/aspnetcore/blob/c85baf8db0c72ae8e68643029d514b2e737c9fae/src/Hosting/Hosting/src/StaticWebAssets/StaticWebAssetsLoader.cs#L35-L43) to ensure the RCL static assets are served from `_content/{PACKAGE_ID}/`, without having to copy files into your project (which clutters the project, might get updated, deleted, etc.). When running `dotnet publish`, an MSBuild target copies the RCL static assets to your output directory, so it also works on non-development environments (and without requiring the `CompositeFileProvider`).

**To be very explicit about this: RCLs are supported on ASP.NET Core 6 out of the box and this PR _does_ break existing behavior!**

I've updated the sample project to .NET 6, added both an embedded file and RCL project reference on the [`feature/rcl-support`](https://github.com/ronaldbarendse/ImageSharp.Web/tree/feature/rcl-support) branch, so you can test this yourself. And you can also see it working again by uncommenting the [`.AddProvider<WebRootImageProvider>`](https://github.com/ronaldbarendse/ImageSharp.Web/blob/c0669a70f4929f875aee32a172252df66413b62b/samples/ImageSharp.Web.Sample/Startup.cs#L44) (which uses the existing [`PhysicalFileSystemResolver` by the way](https://github.com/ronaldbarendse/ImageSharp.Web/blob/c0669a70f4929f875aee32a172252df66413b62b/src/ImageSharp.Web/Providers/WebRootImageProvider.cs#L69)).

I just can't wrap my head around why you did include a `LegacyV1CacheKey` implementation purely for backwards compatibility with V1 users, but don't want to include a `WebRootImageProvider` for the exact same reason (to allow switch back to the previous behavior). I know writing this provider is trivial, but why not include it by default then? And by doing so, highlighting the weird naming of the `PhysicalFileSystemResolver` again 😉

Reply:
It doesn’t break behaviour in a manner that cannot be easily fixed via configuration. If you are storing images outside of the web root you have to configure the static handler. Which most people simply won’t be doing. The provider is command only anyway. 

I HAD to include the legacy cache key as ALL of the replacements break all caches. That will mean silently breaking millions of cached images for some. That’s a profound difference. 

As to why I don’t add one myself. Why SHOULD I? What incentive do I have to add and maintain MORE code with NO BENEFIT to myself.

You need to understand. The maintenance burden for these libraries is MASSIVE. There’s very few bodies of work out there as complicated as the Six Labors libraries maintained by so few. That’s why this middleware was specifically designed to be easily extensible. I give you (figuratively) the tools to easily build items for your requirements and your (figuratively) own paid time.

Reply:
The resolver is not oddly named. It is the resolver for the physical provider. They come in pairs, I explained this already.

Reply:
> As to why I don’t add one myself. Why SHOULD I? What incentive do I have to add and maintain MORE code with NO BENEFIT to myself.
> 
> You need to understand. The maintenance burden for these libraries is MASSIVE. There’s very few bodies of work out there as complicated as the Six Labors libraries maintained by so few. That’s why this middleware was specifically designed to be easily extensible. I give you (figuratively) the tools to easily build items for your requirements and your (figuratively) own paid time.

I totally understand your view on this, but you're actually introducing more code to maintain and making it harder to configure correctly. ASP.NET Core already configures a `PhysicalFileProvider` based on the `WebRootPath`, but only when that folder exists: https://github.com/dotnet/aspnetcore/blob/c85baf8db0c72ae8e68643029d514b2e737c9fae/src/Hosting/Hosting/src/Internal/HostingEnvironmentExtensions.cs#L31-L58.

So I have the following major issues with the current implementation:
1. If the `wwwroot` directory doesn't exist, the `WebRootPath` is not set (and the `WebRootFileProvider` is set to a `NullFileProvider` instance) and this would currently throw exceptions in both `PhysicalFileSystemProvider` and `PhysicalFileSystemCache`;
2. If the exceptions would be fixed, the current `PhysicalFileSystemProvider` creates the missing directory, which would result in a different `WebRootFileProvider` to be assigned on the next startup. There's also no reason for requiring write-access to create the directory, as this provider is read-only (it makes sense for `PhysicalFileSystemCache` to create the cache directory though, as that _does_ require write-access);
3. If you'd fix both these issues, you'll end up with almost the same code as in ASP.NET Core (an thus have introduced additional code to maintain);
4. And because this provider command only, end-users would indeed need to configure additional static file serving to also make the original files available from a directory outside of the web root, which makes it harder to setup correctly. This would again increase the maintenance support, as it might require additional support and/or documentation.

I hope you also understand that I'm trying to help improve the OOTB experience of this library, so everyone can benefit from the time I've put into this, instead of having to (re-)implement core functionality outside of this library myself. No hard feelings though, as you're the maintainer and have the final say, but just want to express my intentions/views on this as well!

Reply:
1. This is by design. If a folder is not explicitly set and the default is not set then it should through. You need to stop coupling the  provider/cache mentally with the ASP.NET Core providers. They're separate and distinct and work well out of the box. (Removing wwwroot is not a common activity)
2. This is a scenario that can never exist because we do and should throw.
3. They're not issues. See comment above about mental coupling.
4. See comment above about mental coupling.

What I think we should do actually. Is strip out all references to `PhysicalFileProvider ` in both the provider and cache and simply use `File`. That way they are obviously distinct and we can move on.